stdlib/public/Observation/Sources/Observation/Observable.swift

```diff
@@ -9,49 +9,28 @@
 //
 //===----------------------------------------------------------------------===//
 
-import _Concurrency
 
 @available(SwiftStdlib 5.9, *)
-public protocol Observable {
-  nonisolated func changes<Isolation: Actor>(
-    for properties: TrackedProperties<Self>,
-    isolatedTo isolation: Isolation
-  ) -> ObservedChanges<Self, Isolation>
-  
-  nonisolated func values<Member: Sendable>(
-    for keyPath: KeyPath<Self, Member>
-  ) -> ObservedValues<Self, Member>
-  
-  nonisolated static func dependencies(
-    of keyPath: PartialKeyPath<Self>
-  ) -> TrackedProperties<Self>
-}
+@_marker public protocol Observable { }
```

Comment:
the problem with normal protocol adoption is that it would break subclassing. the marker is for utility to APIs that want to restrict to something that has been marked as observable.

If later we want to add in protocol based methods/requirements we would need to have finer grained constraints (e.g. identifying the inheritance etc)

Reply:
I don't understand how this breaks subclassing. I think we're going to find the marker protocol to be very limiting

Reply:
so the problem with normal protocols would be that if the type is marked as `@Observable` then the macro generates a conformance via an extension: `extension MyType: Observable { }`. Then if `MySubclassType` inherits from `MyType` and also gets a `@Observable` application to it (which makes sense for observation being required on each layer since the macro does not have knowledge beyond what is typed directly on the type) it means that there will be an `extension MySubclassType: Observable { }`. In the case of normal protocols that emits an error of a redundant protocol conformance.

Marker protocols do not have this restriction.

I feel that if we have additional protocol emissions for specific behaviors in the future we can add a new (non marker) protocol that has the desired features we want.